Makefile

```diff
@@ -63,98 +62,85 @@ ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
 
-# Build manager binary
-build: generate fmt vet
+build: generate fmt vet ## Build manager binary
 	$(build_cmd)
 
-dist: generate fmt vet
+dist: generate fmt vet ## Generates a release
 	goreleaser release --snapshot --rm-dist --skip-sign
 
-# Run against the configured Kubernetes cluster in ~/.kube/config
-run: fmt vet
+run: fmt vet ## Run against the configured Kubernetes cluster in ~/.kube/config
 	go run ./main.go
 
-# Install CRDs into a cluster
-install: generate
+install: generate ## Install CRDs into a cluster
 	$(KUSTOMIZE) build $(CRD_ROOT_DIR)/v1 | kubectl apply -f -
 
-# Uninstall CRDs from a cluster
-uninstall: generate
+uninstall: generate ## Uninstall CRDs from a cluster
 	$(KUSTOMIZE) build $(CRD_ROOT_DIR)/v1 | kubectl delete -f -
 
-# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
-deploy: generate
+deploy: generate ## Deploy controller in the configured Kubernetes cluster in ~/.kube/config
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
 
-# Generate manifests e.g. CRD, RBAC etc.
-generate:
+generate: ## Generate manifests e.g. CRD, RBAC etc.
 	@CRD_ROOT_DIR="$(CRD_ROOT_DIR)" go generate -tags=generate generate.go
 	@rm config/*.yaml
 
-# Generate CRD to file
-crd: generate
+crd: generate ## Generate CRD to file
 	$(KUSTOMIZE) build $(CRD_ROOT_DIR)/v1 > $(CRD_FILE)
 	$(KUSTOMIZE) build $(CRD_ROOT_DIR)/v1beta1 > $(CRD_FILE_LEGACY)
 
-# Run go fmt against code
-fmt:
+fmt: ## Run go fmt against code
 	go fmt ./...
 
-# Run go vet against code
-vet:
+vet: ## Run go vet against code
 	go vet ./...
 
-lint: fmt vet
+lint: fmt vet ## Invokes the fmt and vet targets
 	@echo 'Check for uncommitted changes ...'
 	git diff --exit-code
 
 # Build the binary without running generators
 $(BIN_FILENAME):
 	$(build_cmd)
 
-# Build the docker image
-docker-build: $(BIN_FILENAME)
+docker-build: $(BIN_FILENAME) ## Build the docker image
 	docker build . -t $(DOCKER_IMG) -t $(QUAY_IMG) -t $(E2E_IMG)
 
-# Push the docker image
-docker-push:
+docker-push: ## Push the docker image
 	docker push $(DOCKER_IMG)
 	docker push $(QUAY_IMG)
 
-# Generate bundle manifests and metadata, then validate generated files.
 .PHONY: bundle
-bundle: generate
+bundle: generate ## Generate bundle manifests and metadata, then validate generated files.
 	operator-sdk generate kustomize manifests -q
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
 	operator-sdk bundle validate ./bundle
 
-docs-serve:
+docs-serve: ## Locally run the docs server
 	$(antora_preview_cmd)
 
-install_bats:
+install_bats: ## Installs the bats util via NPM
 	$(MAKE) -C e2e install_bats
 
-e2e_test: docker-build
+e2e_test: docker-build ## Runs the e2e test suite
```

Comment:
```suggestion
e2e_test: docker-build ## Runs the e2e test suite (requires bats, see install_bats)
```

Reply:
IMO sollte das Makefile ja selber merken, dass `install_bats` noch nicht gelaufen ist. Das habe ich nun so umgesetzt.

Reply:
I generally agree, but then it should also depend on `setup_e2e_test`, but both these targets take a bit of time. So to make local development easier (when running over and over again) I decided to separate those

Reply:
The most correct approach would be to make Make skip when kind and bats are set up. But I couldn't come up with an easy solution without overengineering, so I gave up and accepted a little imperfection :)

Reply:
A tactic to work around that in Makefiles is to create marker files. I.e. one would `touch testfiles/.setup_e2e_test` at the end of the respective target, which avoids the target from being run all over again when it ran already. Shall I add this?

Reply:
Let's try, but I would recommend `testbin/` dir instead, to reuse an existing one.

Reply:
That's what I meant 🙈